src/providers/postgres/qgspostgresconn.cpp

```diff
@@ -1686,12 +1686,14 @@ void QgsPostgresConn::retrieveLayerTypes( QVector<QgsPostgresLayerProperty *> &l
 
     if ( layerProperty.isRaster )
     {
-      QString sql = QStringLiteral( "SELECT %3, "
-                                    "array_agg(DISTINCT ST_SRID( %1 ) || ':RASTER')"
-                                    " FROM %2" )
-                    .arg( quotedIdentifier( layerProperty.geometryColName ) )
-                    .arg( table )
-                    .arg( i - 1 );
+      const QString sql = QStringLiteral( "SELECT %1, "
+                                          "array_agg( srid || ':RASTER') "
```

Comment:
take into account that potencially srid value can be different from the real one of the raster

Reply:
I know it can happen, but fetching all the (potentially thousands) raster tiles just to get the SRID really kills performances. And I don't think we can handle the case of multiple raster SRIDs in the same table, at least not with current provider implementations (both GDAL and native).